specs/004-enhance-the-casting/quickstart.md

```
@@ -0,0 +1,438 @@
+// u32 → char: Requires validation (exclude surrogates, >0x10FFFF)
+let rule = matrix.get_promotion_rule(&IrType::U32, &IrType::Char).unwrap();
+assert_eq!(rule.cast_kind, CastKind::IntToChar);
+assert_eq!(rule.requires_validation, true);  // Runtime check needed!
+
+// Invalid Unicode code points:
+// - U+D800 to U+DFFF (surrogates, reserved for UTF-16)
+// - > U+10FFFF (beyond Unicode range)
+```
+
+**Unicode Validation Logic**:
+```rust
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)